test/input-test.js

```diff
@@ -0,0 +1,220 @@
+    expect(component.find('span').hasClass('input-group-addon input-group-addon-primary addon-right')).to.equal(true);
+    expect(component.find('span.addon-right').text()).to.equal('testing right');
+  });
+
+  it('Is addonLeft rendering icons', function() {
+    const icon = <i className='fa fa-check' />;
+    const component = shallow(<Input onChange={noop} addonLeft={<i className='fa fa-check' />} validationResult={true} type='text'/>);
+    expect(component.find('span').containsMatchingElement(icon)).to.equal(true);
+  });
+
+  it('Is addonRight rendering icons', function() {
+    const icon = <i className='fa fa-check' />;
+    const component = shallow(<Input onChange={noop} addonRight={<i className='fa fa-check' />} validationResult={true} type='text'/>);
+    expect(component.find('span').containsMatchingElement(icon)).to.equal(true);
+  });
+
+  it('Shows success state', function() {
+    const component = shallow(<Input onChange={noop} onValidate={() => true} />);
+    expect(component.find('div.form-group').hasClass('has-success')).to.equal(true);
+    expect(component.find('i.ion-checkmark-circled').length).to.equal(1);
+  });
+
+  it('Shows success state without icon', function() {
+    const component = shallow(<Input onChange={noop} onValidate={() => true} hasFeedbackIcon={false} />);
+    expect(component.find('div.form-group').hasClass('has-success')).to.equal(true);
+    expect(component.find('i').length).to.equal(0)
+  });
+
+  it('Shows fail state', function() {
+    const component = shallow(<Input onChange={noop} onValidate={() => false} />);
+    expect(component.find('div.form-group').hasClass('has-error')).to.equal(true);
+    expect(component.find('i.ion-android-cancel').length).to.equal(1);
+  });
+
+  it('Shows warning state', function() {
+    const component = shallow(<Input onChange={noop} onValidate={() => 'warning'} />);
+    expect(component.find('div.form-group').hasClass('has-warning')).to.equal(true);
+    expect(component.find('i.ion-alert-circled').length).to.equal(1);
+  });
+
+  it('Shows addon Right', function() {
+    const component = shallow(<Input onChange={noop} addonRight='testing right' onValidate={() => true} />);
+    expect(component.find('span').hasClass('input-group-addon input-group-addon-primary addon-right')).to.equal(true);
+    expect(component.find('span.addon-right').text()).to.equal('testing right');
+  });
+
+  it('Shows addon Left', function() {
+    const component = shallow(<Input onChange={noop} addonLeft='testing left' onValidate={() => true} />);
+    expect(component.find('span').hasClass('input-group-addon input-group-addon-primary addon-left')).to.equal(true);
+    expect(component.find('span.addon-left').text()).to.equal('testing left');
+  });
+
+  it('Has an id/key');
+// Key is not a standard prop, cannot access for testing (https://fb.me/react-special-props)
+
+  it('onValidate is called', function() {
+    const onClick = sinon.spy();
```

Comment:
Since we can call this variable anything and it isn't triggering an onClick even, can we call this onValidate?